#1174 More consistent palette import behavior

davidlamhauge opened the pull request:
![David_Pencil2d_20190911_20_20_22](https://user-images.githubusercontent.com/1292931/64724524-f16a3180-d4d2-11e9-83c5-221513069ebf.gif)
This PR addresses the inconsistent behavior on opening, importing, saving and exporting palettes.
As we agreed, this is how it goes:
1) Opening a palette, replaces the existing palette with a new one. If at least one color is in use on a vector layer, a warning is given. Only one warning.
2) Importing a palette, appends the new palette to the old one.
3) Saving a palette can be done to the Pencil2D xml format.
4) Exporting a palette can be done to GPL, XML, and other formats as soon as the formats are implemented.
I hope I've covered all (or most) issues we've talked about.
This closes #1174

CandyFace commented on app/src/mainwindow2.cpp:
```diff
@@ -1295,6 +1297,54 @@ void MainWindow2::importPalette()
+{
+    int count = 0;
+    int maxNumber = mEditor->object()->getColourCount();
+    bool openPalet = true;
```
typo, should be openPalette. It's a minor thing but makes it much easier to search.

CandyFace commented on app/ui/mainwindow2.ui:
```diff
@@ -81,6 +82,7 @@
      <addaction name="actionExport_Animated_GIF"/>
      <addaction name="separator"/>
      <addaction name="actionExport_Palette"/>
+     <addaction name="actionExport_Other_Palette_formats"/>
```
I don't see a need to have multiple export actions, just provide one "Export Palette" then let the dialog. Remove the Export palette and keep the dialog where you can select extension.

CandyFace commented on core_lib/src/structure/object.cpp:
```diff
@@ -516,7 +516,7 @@ void Object::importPalettePencil(QFile& file)
     doc.setContent(&file);
 
-    mPalette.clear();
+//    mPalette.clear();
```
This should be removed

davidlamhauge commented:
I have merged the newest Master branch into this branch, and fixed the typs.
I also removed the option to choose whether to export as Pencil2D or GIMP format. It works, but on my Ubuntu you must add suffix yourself.
I'm too tired to fix it today... :sleepy:

davidlamhauge commented:
Now, there is always an extension.

davidlamhauge commented:
It again saves without file extension in Linux (Ubuntu 18.04).
Otherwise it works fine @CandyFace

CandyFace commented:
The save without extension on linux should be fixed via https://github.com/pencil2d/pencil/pull/1299

CandyFace commented:
I'd say this is ready to be merged whenever the remaining conflicts has been fixed.

davidlamhauge commented:
Conflicts has been fixed.

CandyFace commented on core_lib/src/util/fileformat.h:
```diff
@@ -63,6 +63,8 @@ GNU General Public License for more details.
 #define PFF_DEFAULT_PALETTE_EXT \
     QString(".xml")
 
+#define PFF_PALETTE_EXT_FILTER \
```
This define is overriding the one from master and should be removed. It's redundant since the macro is defined above.

davidlamhauge replied:
Removed
